chemprop/featurizers/molecule.py

```diff
@@ -41,3 +49,68 @@ class MorganBinaryFeaturizer(MorganFeaturizerMixin, BinaryFeaturizerMixin, Vecto
 @MoleculeFeaturizerRegistry("morgan_count")
 class MorganCountFeaturizer(MorganFeaturizerMixin, CountFeaturizerMixin, VectorFeaturizer[Mol]):
     pass
+
+
+@MoleculeFeaturizerRegistry("rdkit_2d")
+class RDKit2DFeaturizer(VectorFeaturizer[Mol]):
+    def __init__(self):
+        warnings.warn(
+            "The RDKit 2D features can deviate signifcantly from a normal distribution. Consider "
+            "manually scaling them using an appropriate scaler before creating datapoints, rather "
+            "than using the scikit-learn `StandardScaler` (the default in Chemprop)."
+        )
+
+    def __len__(self) -> int:
+        return len(Descriptors.descList)
+
+    def __call__(self, mol: Chem.Mol) -> np.ndarray:
+        features = np.array(
+            [
+                func(mol)
+                for _, func in filter(
+                    lambda i: i[0] != "SPS" or mol.GetNumHeavyAtoms() > 0, Descriptors.descList
+                )
+            ],
+            dtype=float,
+        )
+
+        return features
+
+
+if not NO_DESCRIPTASTORUS:
+
+    class V1RDKit2DFeaturizerMixin(VectorFeaturizer[Mol]):
```

Comment:
What is the difference between `RDKit2DFeaturizer` and `V1RDKit2DFeaturizer`? Should we just keep one of them to avoid confusion?

Reply:
It seems the new one has 10 more descriptors. Maybe we can just keep one of them.
> 'AvgIpc', 'BCUT2D_CHGHI', 'BCUT2D_CHGLO', 'BCUT2D_LOGPHI', 'BCUT2D_LOGPLOW', 'BCUT2D_MRHI', 'BCUT2D_MRLOW', 'BCUT2D_MWHI', 'BCUT2D_MWLOW', and 'SPS'

Reply:
If we had to keep just one, I would recommend it being the V1RDKit2DFeaturizer because if someone is trying to use a v1 model that used rdkit features, they would need that featurizer. 

I added a new version because of the extra descriptors that have been added since descriptastorus. It could be nice to let people get these more descriptors. My intention is that all new models would use RDKit2DFeaturizer and that the V1 version would only be used for models that were trained in v1. 

Let me know if you feel strongly, or we can bring it up in the next meeting.

Reply:
I am okay with keeping both of them. However, in the long term, I don't think people will care about V1RDKit2DFeaturizer anymore, as it will become a legacy we used in the past. If someone wants to do a benchmark with this smaller set, they should take responsibility for generating those features correctly via a notebook and provide them as additional features.